cgo/lapack.go

```diff
@@ -413,6 +425,63 @@ func (impl Implementation) Dlarfx(side blas.Side, m, n int, v []float64, tau flo
 	lapacke.Dlarfx(side, m, n, v, tau, c, ldc, work)
 }
 
+// Dlascl multiplies an m×n matrix by the scalar cto/cfrom.
+//
+// cfrom must not be zero, and cto and cfrom must not be NaN, otherwise Dlascl
+// will panic.
+//
+// Dlascl is an internal routine. It is exported for testing purposes.
+func (impl Implementation) Dlascl(kind lapack.MatrixType, kl, ku int, cfrom, cto float64, m, n int, a []float64, lda int) {
+	checkMatrix(m, n, a, lda)
+	if cfrom == 0 {
+		panic(zeroDiv)
+	}
+	if math.IsNaN(cfrom) || math.IsNaN(cto) {
+		panic(nanScale)
+	}
+	lapacke.Dlascl(byte(kind), kl, ku, cfrom, cto, m, n, a, lda)
+}
+
+// Dlaset sets the off-diagonal elements of A to alpha, and the diagonal
```

Comment:
The change in #216 doesn't match the Fortran docs and is more invasive. The documented [behaviour](http://www.netlib.org/cgi-bin/netlibfiles.txt?format=txt&filename=/lapack/lapack_routine/dlacpy.f) of ?lacpy is to use all when not U or L. I would do what is suggested above.

```
*> DLACPY copies all or part of a two-dimensional matrix A to another
*> matrix B.
*> \endverbatim
*
*  Arguments:
*  ==========
*
*> \param[in] UPLO
*> \verbatim
*>          UPLO is CHARACTER*1
*>          Specifies the part of the matrix A to be copied to B.
*>          = 'U':      Upper triangular part
*>          = 'L':      Lower triangular part
*>          Otherwise:  All of the matrix A
*> \endverbatim
```